projects/uniswap-v1/index.js

```diff
@@ -1,13 +1,31 @@
-const { getChainTvl } = require('../helper/getUniSubgraphTvl');
+const { nullAddress } = require('../helper/unwrapLPs')
+const sdk = require("@defillama/sdk")
 
-const v1graph = getChainTvl({
-  ethereum: 'ESnjgAG9NjfmHypk4Huu4PVvz55fUwpyrRqHF21thoLJ'
-}, "uniswaps", "totalLiquidityUSD")
+const uniFactory = '0xc0a47dFe034B400B47bDaD5FecDa2621de6c4d95'
+
+const abi = {
+  getExchange: "function getExchange(address token) view returns (address)",
+  getTokenWithId: "function getTokenWithId(uint256 token_id) view returns (address)",
+  tokenCount: "function tokenCount() view returns (uint256)",
+}
+
+const tvl = async (api) => {
+  const tokens = (await api.fetchList({ lengthAbi: abi.tokenCount, itemAbi: abi.getTokenWithId, target: uniFactory })).filter((token) => token !== nullAddress)
+  const pools = await api.multiCall({ calls: tokens.map((token) => ({ target: uniFactory, params: [token] })), abi: abi.getExchange })
+  const balanceOfs = await api.multiCall({ calls: pools.map((pool, i) => ({ target: tokens[i], params: [pool] })), abi: 'erc20:balanceOf', permitFailure: true })
+
+  await Promise.all(
```

Comment:
think this is part of the issue, there is 4k pools and then maybe even if 1/5th have some token in it, this is too many parallel calls against the rpc